### PR TITLE
Refactored handling of JS AST to be recursive

### PR DESCRIPTION
**Pull request by swalsh15**
Refactoring JS to use router files and recursion to parse AST tree. Supports variable declarations and console.log

### CODE REVIEW

File: translate/js_main.py [hunk 1]
```diff
@@ -7,17 +7,15 @@
...
 sys.path.append('routers')
 
-import js_router as router
+import js_router
 
-program = "let my_container = 20\n console.log('eh')"
+program = "let x = 's' + 1 + 3\n console.log(x)"
 
+"""
+takes js string and converts it to a general AST
```
Comment (augerm) [inline translate/js_main.py:15]: `general` --> `generic`

File: translate/js_main.py [hunk 2]
```diff
@@ -7,17 +7,15 @@
 sys.path.append('helpers')
 sys.path.append('routers')
 
-import js_router as router
+import js_router
 
-program = "let my_container = 20\n console.log('eh')"
+program = "let x = 's' + 1 + 3\n console.log(x)"
```
Comment (augerm) [inline translate/js_main.py:12]: Impressive :)

File: translate/routers/js_router.py [hunk 3]
```diff
@@ -1,16 +1,38 @@
-import js_assign as a
```
Comment (augerm) [inline translate/routers/js_router.py:1]: Thanks!

File: translate/expression/js_expression.py [hunk 4]
```diff
@@ -1,10 +1,10 @@
 import js_helpers as h
```
Comment (augerm) [inline translate/expression/js_expression.py:1]: Can you remove this alias while you're here?

File: translate/expression/js_expression.py [hunk 5]
```diff
@@ -20,3 +20,28 @@ def jsexpr_to_gast(node):
         # handle args
         gast["args"] = h.jsargs_to_strlist(node.arguments)
     return gast
+
+def expr(node):
```
Comment (augerm) [inline translate/expression/js_expression.py:24]: Maybe rename to `def convert_expression_to_gast()`?

File: translate/expression/js_expression.py [hunk 6]
```diff
@@ -20,3 +20,28 @@ def jsexpr_to_gast(node):
...
+    return js_router.node_to_gast(node.expression)
+
+"""
+takes python ast call node and converts to generic ast format
+example print('hello'):
+    exampleIn Call(func=Name(id='print'), args=[Str(s='hello')], keywords=[])
+    exampleOut {'type': 'logStatement', 'args': ['hello']}
+"""
+def call(node):
```
Comment (augerm) [inline translate/expression/js_expression.py:33]: Same here rename `def call(node)` --> `def call_expression_to_gast()` or something more explicit.

File: translate/helpers/js_helpers.py [hunk 7]
```diff
@@ -16,6 +17,18 @@ def jsarg_to_str(arg):
...
     return ""
 
+"""
+takes ast node of type program and returns
+a generic ast for that node
+example print("hello"):
+    node (input): Program(body=[ExpressionStatement(value=Call(func=Name(id='print'), args=[Str(s='hello')], keywords=[]))])
+    gast (output): {'type': 'root', 'body': [{'type': 'logStatement', 'args': ['hello']}]}
+"""
+def program(node):
```
Comment (Danish21) [inline translate/helpers/js_helpers.py:27]: same rename to program_to_gast() and similiar for all functions.

**Pull request merged**